### PR TITLE
Separated the logic from the view

### PR DESCRIPTION
**Pull request by JunaidRajah**
In preparation of integrating unit tests I separated the logic from the view and added a few changes mentioned in the code review.

### CODE REVIEW

**Comment by JunaidRajah** [comment]
Added Unit Tests for the Logic

File: guesstheflagTests/guesstheflagTests.swift [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+//
+
+import XCTest
+
+class guesstheflagTests: XCTestCase {
+    
+    let flagLogic = FlagLogic()
+    
+    override func setUpWithError() throws {
+        // Put setup code here. This method is called before the invocation of each test method in the class.
```
Comment (tmanOC) [inline guesstheflagTests/guesstheflagTests.swift:15]: Rather use a new FlagLogic before each test. To help keep all your tests Isolated/Independent.
So change flagLogic to a var, and add the initialising in the setup method.

File: guesstheflag/FlagLogic.swift [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        score += isCorrect ?  1 : -1
+        correctFlags = correctFlags + (userAnswer == correctAnswer ?  1 : 0)
+        incorrectFlags = incorrectFlags + (userAnswer == correctAnswer ?  0 : 1)
+        return isCorrect
+    }
+    
+    func hasGameCompleted() -> Bool{
+        let gameComplete =  (questionCount >= 9 ? true : false)
+        questionCount = questionCount >= 9 ? 0: questionCount
+        return gameComplete
```
Comment (tmanOC) [inline guesstheflag/FlagLogic.swift:81]: I think it is good to split this method up because it is not clear from the name of the message that it also resets the questionCount when the game is over

I think it is a good idea to add a test for this hasGameCompleted method.

File: guesstheflagTests/guesstheflagTests.swift [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    
+    func testGetCorrectCountryIsValid() {
+        let correctCountry = flagLogic.getCorrectCountryName
+        let noOutput = ""
+        XCTAssertNotEqual(correctCountry, noOutput)
+    }
+    
+    func testCorrectAnswerIncrementsScoreBy1() {
+        let currentScore = flagLogic.score
+        let _ = flagLogic.isCorrectAnswer(userAnswer: flagLogic.getCorrectCountryPosition)
```
Comment (tmanOC) [inline guesstheflagTests/guesstheflagTests.swift:60]: I think it's also good to add a test for when isCorrectAnswer receives out of range selections

**Review by tmanOC — CHANGES_REQUESTED** [review]
